lightweight_lidar_only_simulator:
  ros__parameters:

    # The distance between the front and
    # rear axle of the racecar
    wheelbase: 0.325 # meters

    # The period of publication (in seconds) of pose and lidar datta
    update_pose_period: 0.02

    # Lidar simulation parameters
    scan_beams: 100
    scan_field_of_view: 4.71 # radians

    # The standard deviation of the noise applied
    # to the lidar simulation
    scan_std_dev: 0.01 # meters

    # The probability threshold for points
    # in the occupancy grid to be considered "free".
    # Used for the lidar simulator.
    map_free_threshold: 0.3

    # The distance from the center of the
    # rear axis (base_link) to the lidar
    scan_distance_to_base_link: 0.275 # meters
    
    # Limits on the speed and steering angle
    max_speed: 4. # meters/second
    max_steering_angle: 0.34 # radians

    # Enables joystick if true
    joy: true
    joy_speed_axis: 1
    joy_angle_axis: 2
    joy_max_speed: 2.0 # meters/second

    # The names of topics to listen and publish to
    joy_topic: "/joy"
    drive_topic: "/ackermann_drive_commands"
    map_topic: "/map"
    joint_states_topic: "/joint_states"
    scan_topic: "/scan"
    pose_topic: "/pose"
    pose_rviz_topic: "/initialpose"
    odom_topic: "/odom"
    distance_transform_topic: "/dt"

    # The names of the transformation frames published to
    map_frame: "map"
    base_frame: "base_link"
    scan_frame: "laser"

    # Turn this off for mapping
    broadcast_transform: true
